tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java

```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
 	public int getTimeout() {
 		return timeout;
 	}
+	
+	/** @see #setEnableImageProcessing(boolean) */
+	public int isEnableImageProcessing() {
+		return enableImageProcessing;
+	}
+
+	/** 
+	 * Set the value to true if processing is to be enabled.
+	 * Default value is false.
+	 */
+	public void setEnableImageProcessing(int enableImageProcessing) {
+		this.enableImageProcessing = enableImageProcessing;
+	}
+
+	/**
+	 * @return the density
+	 */
+	public int getDensity() {
+		return density;
+	}
+
+	/**
+	 * @param density the density to set
+	 * Default value is 300.
+	 */
+	public void setDensity(int density) {
+		if(density < 150 || density > 1200) {
+			throw new IllegalArgumentException("Invalid density value");
+		}
+		this.density = density;
+	}
 
 	/**
+	 * @return the depth
+	 */
+	public int getDepth() {
+		return depth;
+	}
+
+	/**
+	 * @param depth the depth to set
+	 * Default value is 4.
+	 */
+	public void setDepth(int depth) {
+		int[] allowedValues = {2, 4, 8, 16, 32, 64, 256, 4096};
+		for (int i = 0; i < allowedValues.length; i++) {
+			if(depth == allowedValues[i]) {
+				this.depth = depth;
+				return;
+			}
+		}
+		throw new IllegalArgumentException("Invalid depth value");
+	}
+
+	/**
+	 * @return the colorspace
+	 */
+	public String getColorspace() {
+		return colorspace;
+	}
+
+	/**
+	 * @param colorspace the colorspace to set
+	 * Deafult value is gray.
+	 */
+	public void setColorspace(String colorspace) {
+		if(!colorspace.equals(null)) {
+			this.colorspace = colorspace;
+		} else {
+			throw new IllegalArgumentException("Invalid colorspace value");
```

Comment:
This Exception message is misleading. The value is not invalid... it is simply null. The message should state "Colorspace value cannot be null"